crates/pathfinder/src/rpc/types.rs

```diff
@@ -571,24 +571,47 @@ pub mod reply {
             let txn = txn
                 .transaction
                 .ok_or_else(|| anyhow::anyhow!("Transaction not found."))?;
-            Ok(Self {
-                txn_hash: txn.transaction_hash,
-                contract_address: txn.contract_address,
-                entry_point_selector: txn.entry_point_selector,
-                calldata: txn.calldata,
-                max_fee: txn.max_fee,
+
+            Ok(match txn {
+                sequencer::reply::transaction::Transaction::Invoke(txn) => Self {
+                    txn_hash: txn.transaction_hash,
+                    contract_address: Some(txn.contract_address),
+                    entry_point_selector: Some(txn.entry_point_selector),
+                    calldata: Some(txn.calldata),
+                    max_fee: Some(txn.max_fee),
+                },
+                _ => Self {
+                    // TODO this is probably the best we can do until the RPC spec introduces
+                    // 3 variants for all the transaction types
+                    txn_hash: txn.hash(),
+                    contract_address: None,
+                    entry_point_selector: None,
+                    calldata: None,
+                    max_fee: None,
```

Comment:
`max_fee` is available in `Declare`, while `contract_address` is available in `Deploy`, yet I think we should just populate the transaction hash for now, until the RPC spec is updated.

Reply:
Could you elaborate what is the connection of first and second with the third?
1. `max_fee in Declare`
2. `contract_address in Deploy`
3. just populate `Transaction::txn_hash` for now

Reply:
Honestly it was just an arbitrary choice of mine, which I am happy to change to whatever you suggest.

The problem here is that the output (ie. rpc type) does not fit the real data that is available in 3 variants. The spec needs an update, yet we need to plug this hole temporarily.